core/src/test/scala/org/apache/spark/util/AccumulatorV2Suite.scala

```diff
@@ -86,4 +86,63 @@ class AccumulatorV2Suite extends SparkFunSuite {
     assert(acc.sum == 5.0)
     assert(acc.avg == 1.25)
   }
+
+  test("ListAccumulator") {
+    val acc = new ListAccumulator[Double]
+    assert(acc.value.isEmpty)
+    assert(acc.isZero)
+
+    acc.add(0.0)
+    assert(acc.value.contains(0.0))
+    assert(!acc.isZero)
+
+    acc.add(new java.lang.Double(1.0))
```

Comment:
we probably don't need this since ListAccumulator does not differentiate specialized vs non-specialized.

but it's minor so i'm going to just merge this.